src/Filament/Widgets/Components.php

```diff
@@ -2,11 +2,78 @@
+use Filament\Forms\Components\Component as FilamentFormComponent;
+use Filament\Forms\Components\Group;
+use Filament\Forms\Components\Radio;
+use Filament\Forms\Components\Section;
+use Filament\Forms\Concerns\InteractsWithForms;
+use Filament\Forms\Contracts\HasForms;
+use Filament\Forms\Form;
 use Filament\Widgets\Widget;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Support\Collection;
 
-class Components extends Widget
+class Components extends Widget implements HasForms
 {
+    use InteractsWithForms;
+
     protected static string $view = 'cachet::filament.widgets.components';
 
     protected int|string|array $columnSpan = 'full';
+
+    public Collection $formData;
+    public  Collection $componentGroups;
+
+    public function mount(): void
+    {
+        $this->componentGroups = ComponentGroup::query()
+            ->select(['id', 'name', 'collapsed', 'visible'])
+            ->where('visible', '=', true)
+            ->with('components', fn (HasMany $query) => $this->loadComponents($query))
+            ->get();
+
+        $this->formData = $this->componentGroups->mapWithKeys(function (ComponentGroup $componentGroup) {
+            $components = $componentGroup->components->mapWithKeys(function (Component $component) {
+                return [$component->id => $component->only('status')];
+            });
+            return [$componentGroup->id => ['components' => $components]];
+        });
+    }
+
+    public function form(Form $form): Form
+    {
+        $schema = $this->componentGroups
+            ->loadMissing(['components' => fn (HasMany $query) => $this->loadComponents($query)])
+            ->map(function (ComponentGroup $componentGroup): FilamentFormComponent {
+                return Section::make($componentGroup->name)
+                    ->schema(function () use ($componentGroup) {
+                        return
+                            $componentGroup->components->map(function (Component $component) use ($componentGroup) {
+                                return
+                                    Group::make([
+                                        Radio::make($componentGroup->id . '.components.' . $component->id . '.status')
+                                            ->label($component->name)
+                                            ->columns(count(ComponentStatusEnum::cases()))
+                                            ->live()
+                                            ->options(ComponentStatusEnum::class)
+                                            ->afterStateUpdated(function (ComponentStatusEnum $state) use ($component) {
+                                                return $component->update(['status' => $state]);
+                                            })
+                                    ]);
+                            })->toArray();
+                    })
+                    ->collapsed($componentGroup->isCollapsible());
```

Comment:
Just a hunch, but what happens if you add `->columnSpanFull()` at the end of this? Does this fix your collapsed state issue?

```suggestion
                    ->collapsed($componentGroup->isCollapsible())
                    ->columnSpanFull();
```

Reply:
Yeah good call, I had tried that but it didn't work, then woke up this morning and realised I could just set the form to be full width and it has fixed it 😄